apis/v1alpha2/validation/httproute.go

```diff
@@ -166,3 +143,35 @@ func validateHTTPPathMatch(path *gatewayv1a2.HTTPPathMatch, fldPath *field.Path)
 	}
 	return allErrs
 }
+
+// validateHTTPRouteFilterTypeMatchesValue validates that each filter type matches
+//the filter value and that values are not empty.
+func validateHTTPRouteFilterTypeMatchesValue(filter gatewayv1a2.HTTPRouteFilter, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	switch filter.Type {
+	case gatewayv1a2.HTTPRouteFilterExtensionRef:
+		if filter.ExtensionRef == nil || (gatewayv1a2.LocalObjectReference{}) == *filter.ExtensionRef {
+			errs = append(errs, field.Invalid(path.Child("filters"), path, "extensionRef must be specified for ExtensionRef filter"))
+		}
+	case gatewayv1a2.HTTPRouteFilterRequestHeaderModifier:
+		//Request header modifier contains slice whi
+		if filter.RequestHeaderModifier == nil || filter.RequestHeaderModifier.Add == nil || filter.RequestHeaderModifier.Set == nil || filter.RequestHeaderModifier.Remove == nil {
+			errs = append(errs, field.Invalid(path.Child("filters"), path, "requestHeaderModifier must be specified for RequestHeaderModifier filter"))
+		}
+	case gatewayv1a2.HTTPRouteFilterRequestMirror:
+		if filter.RequestMirror == nil || (gatewayv1a2.HTTPRequestMirrorFilter{}) == *filter.RequestMirror {
```

Comment:
```suggestion
		if filter.RequestMirror == nil || gatewayv1a2.HTTPRequestMirrorFilter{} == *filter.RequestMirror {
```

and the other places in this function